src/Metis.jl

```diff
@@ -82,6 +85,40 @@ function graph(G::LightGraphs.AbstractSimpleGraph)
     return Graph(idx_t(N), xadj, adjncy)
 end
 
+"""
+    graph(G::SimpleWeightedGraphs.AbstractSimpleWeightedGraph)
+
+Construct the 1-based CSR representation of the `SimpleWeightedGraphs` graph `G`.
+The edge weights should be positive integers.
+"""
+function graph(G::SimpleWeightedGraphs.AbstractSimpleWeightedGraph)
+    N = SimpleWeightedGraphs.nv(G)
+    xadj = Vector{idx_t}(undef, N+1)
+    xadj[1] = 1
+    adjncy = Vector{idx_t}(undef, 2*SimpleWeightedGraphs.ne(G))
```

Comment:
There is a discrepancy between `SimpleWeightedGraphs.ne()` and `LightGraphs.ne()` because `SimpleWeightedGraphs.ne()` fails to count the correct number of self-loops (please see [here](https://github.com/JuliaGraphs/SimpleWeightedGraphs.jl/issues/41)). The current `Metis.partition()` is able to handle unweighted graphs with self-loops, so I think this part should be fixed to allow self-loops.

Reply:
At least the following implementation avoids the problem.

```julia
adjncy = Vector{idx_t}(undef, 2*(SimpleWeightedGraphs.ne(G)+N))
```